opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -594,21 +594,61 @@ namespace Opm {
 
             auto& ebosJac = ebosSimulator_.model().linearizer().jacobian().istlMatrix();
             auto& ebosResid = ebosSimulator_.model().linearizer().residual();
+            auto& ebosSolver = ebosSimulator_.model().newtonMethod().linearSolver();
 
-            // set initial guess
-            x = 0.0;
+            const int numSolvers = ebosSolver.numAvailableSolvers();
+            if ((numSolvers > 1) && (ebosSolver.getSolveCount() % 100 == 0)) {
 
-            auto& ebosSolver = ebosSimulator_.model().newtonMethod().linearSolver();
-            Dune::Timer perfTimer;
-            perfTimer.start();
-            ebosSolver.prepare(ebosJac, ebosResid);
-            linear_solve_setup_time_ = perfTimer.stop();
-            ebosSolver.setResidual(ebosResid);
-            // actually, the error needs to be calculated after setResidual in order to
-            // account for parallelization properly. since the residual of ECFV
-            // discretizations does not need to be synchronized across processes to be
-            // consistent, this is not relevant for OPM-flow...
-            ebosSolver.solve(x);
+                if ( terminal_output_ ) {
+                    OpmLog::debug("\nRunning speed test for comparing available linear solvers.");
+                }
+
+                Dune::Timer perfTimer;
+                std::vector<double> times(numSolvers);
+                std::vector<double> setupTimes(numSolvers);
+
+                x = 0.0;
+                std::vector<BVector> x_trial(numSolvers, x);
+                for (int solver = 0; solver < ebosSolver.numAvailableSolvers(); ++solver) {
```

Comment:
Tiny nit: `ebosSolver.numAvailableSolvers()` is already available in `numSolvers` above.  It probably doesn't hurt to call that function again, but also doesn't really help&ndash;assuming `numAvailableSolvers()` doesn't change between calls, of course.

Reply:
Changed.